### DISCUSSION TITLE
EVL4 to Home Assistant to HomeKit

### DISCUSSION THREAD

**Discussion by engyew** [ufodone/envisalink_new#64] [General]
Guys, a big thank you for envisalink_new!  I have a Vista 20p with 17 zones, I have a Tuxedo. Using an evl4 and am using HA to bridge to HomeKit.  It works much better than the standard Envisalink integration.  For one, I could never arm my alarm using the standard integration.  

1) Alarm arming works and alarm status is correct. Works in HA *and* Apple Home.
2) I can see my zones in HA as binary sensors.  Thank you!  Cannot see the zones in Apple Home.
3) Just started testing out duress and silent arm. I dont think those works.

Much appreciate the effort! I'll let you know as I proceed.

**Comment by ufodone** [comment]
Hi, thanks for the feedback.  I don't use HomeKit myself so I can't comment on 2 or 3.  If your further investigation looks to be pointing to an issue with this integration, please open an issue and I can try to help debug.

**Comment by COLAXDad** [comment]
@engyew - I am curious does your homekit integration ask for the code to arm/siarm etc?  I have installed today the envisalink_new with my envislink4 for a honeywell Vista 15P.  It works so far in preliminary testing for HA.

However, as it comes over to Homekit; there is no way to enter the code to arm/disarm the system.  I can see all the zones in HA as binary sensors but not in Homekit. It would be helpful to know which zone is "open" preventing the arming.  It would also behelpful to have a bypass available in homekit. 

@ufodone is this a known issue about the accesscode to homekit?  The error I get in the logs is "homeassistant.exceptions.ServiceValidationError: Arming requires a code but none was given for alarm_control_panel.new_alarm_partition_1"

**Reply by engyew** [reply]
No, HomeKit does not ask for a code to disarm.  I am good that it relies on the iPhone built in authentication to validate that you can get to the Home app.  It uses the code you put in in the envisalink_new configuration.  I created a Vista "Arm Only" user to use with Envisalink.  For safety reasons, I do not want my iPhone to disarm my alarm.  I prefer a hands-on disarm.

I can see all my Vista sensors in HomeKit, it takes time to propagate.  I dont use HomeKit for automation, it's too limited.  I use HA.  Best of luck and have fun!

I am running 0.9.1

**Comment by ufodone** [comment]
> is this a known issue about the accesscode to homekit? The error I get in the logs is "homeassistant.exceptions.ServiceValidationError: Arming requires a code but none was given for alarm_control_panel.new_alarm_partition_1"

I'm not aware of any issue here (but that doesn't mean one doesn't exist).  As mentioned, I don't use HomeKit/HomeBridge so don't know how it interacts with HA or how it determines whether a code is required or not.

**Comment by engyew** [comment]
EVL to HA to HomeKit has been working for a year.  Closing
